Sources/Controllers/2.0/OACustomPOIViewController.mm

```diff
@@ -161,79 +336,163 @@ - (void)selectSubcategoryDone:(OAPOICategory *)category keys:(NSMutableSet<NSStr
+- (void)tableView:(UITableView *)tableView willDisplayCell:(UITableViewCell *)cell forRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    if (_searchMode)
+    {
+        OAPOIType *poiType = _searchResult[indexPath.row];
+        NSSet<NSString *> *acceptedSubtypes = [_acceptedTypes objectForKey:poiType.category];
+        BOOL accepted = [acceptedSubtypes containsObject:poiType.name];
+        [cell setSelected:accepted animated:NO];
+        if (accepted)
+            [tableView selectRowAtIndexPath:indexPath animated:NO scrollPosition:UITableViewScrollPositionNone];
         else
-            descText = [NSString stringWithFormat:@"%lu/%lu", countAcceptedTypes, countAllTypes];
-        cell.descriptionView.text = descText;
-        cell.descriptionView.textColor = UIColorFromRGB(color_text_footer);
-
-        [cell updateConstraintsIfNeeded];
+            [tableView deselectRowAtIndexPath:indexPath animated:NO];
     }
-    return cell;
 }
 
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    [tableView deselectRowAtIndexPath:indexPath animated:YES];
+    if (_searchMode)
+    {
+        [self selectDeselectItem:indexPath];
+    }
+    else
+    {
+        [tableView deselectRowAtIndexPath:indexPath animated:YES];
+
+        OAPOICategory* item = _categories[indexPath.row];
+        OASelectSubcategoryViewController *subcategoryScreen = [[OASelectSubcategoryViewController alloc] initWithCategory:item filter:_filter];
+        subcategoryScreen.delegate = self;
+        [self.navigationController pushViewController:subcategoryScreen animated:YES];
+    }
+}
+
+- (void)tableView:(UITableView *)tableView didDeselectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    if (_searchMode)
+        [self selectDeselectItem:indexPath];
+}
 
-    OAPOICategory* item = _categories[indexPath.row];
-    OASelectSubcategoryViewController *subcategoryScreen = [[OASelectSubcategoryViewController alloc] initWithCategory:item filter:_filter];
-    subcategoryScreen.delegate = self;
-    [self.navigationController pushViewController:subcategoryScreen animated:YES];
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 1;
 }
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
 {
-    return _categories.count;
+    return _searchMode ? _searchResult.count : _categories.count;
 }
 
-- (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSInteger)section
+- (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    return [OAPOISearchHelper getHeightForFooter];
+    return _searchMode;
 }
 
-- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
+- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    return [OAPOISearchHelper getHeightForHeader];
+    return _searchMode ? 48 : 66;
```

Comment:
Again, do not implement ```heightForRowAtIndexPath``` under any circumstance except for non-dynamic cells (e.g. picker cells that have a constant height). In the autolayout context all the cell heights are calculated dynamically by the system.